roofit/roofitcore/src/RooRealVar.cxx

```diff
@@ -384,41 +369,46 @@ void RooRealVar::setBins(Int_t nBins, const char* name) {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-/// Add given binning under name 'name' with this variable. If name is null
-/// the binning is installed as the default binning
+/// Add given binning under name 'name' with this variable. If name is null,
+/// the binning is installed as the default binning.
 void RooRealVar::setBinning(const RooAbsBinning& binning, const char* name)
 {
+  std::unique_ptr<RooAbsBinning> newBinning( binning.clone() );
+
   // Process insert hooks required for parameterized binnings
-  if (!name) {
-    RooAbsBinning* newBinning = binning.clone() ;
+  if (!name || strlen(name) == 0) {
```

Comment:
```suggestion
  if (!name || !name[0]) {
```
(No need to search the whole string for `\0` if all you want to know is whether it's empty. I know, a RooFit pattern...)